src/wm/page.rs

```diff
@@ -20,6 +24,32 @@ pub struct Page {
     pub redirects: Vec<Redirect>,
 }
 
+impl Page {
+    pub fn wikidata(&self) -> Option<WikidataQid> {
+        // TODO: return error
+        self.main_entity
+            .as_ref()
+            .map(|e| WikidataQid::from_str(&e.identifier).unwrap())
+    }
+
+    /// Title of the article
+    pub fn title(&self) -> anyhow::Result<WikipediaTitleNorm> {
+        WikipediaTitleNorm::from_title(&self.name, &self.in_language.identifier)
+    }
+
+    /// All titles that lead to the article, the main title followed by any redirects.
+    pub fn all_titles(&self) -> impl Iterator<Item = anyhow::Result<WikipediaTitleNorm>> + '_ {
+        // TODO: return error
```

Comment:
How will it work now?

Reply:
That's an old TODO, I'll remove it. It returns any parse errors it encounters with the title and redirects.